### DISCUSSION TITLE
Refunds on shipping cost

### DISCUSSION THREAD

**Discussion by loicginoux** [solidusio/solidus#4560] [Troubleshooting]
Hi,
We struggle to understand the best way to  refund shipping costs.

At the moment the best way we found is to do it directly from the payment tab, by entering manually the amount of the shipping cost and adding a reason like “refund shipping cost”.  I would like to see  if there is a smoother way for admins to do this.
Maybe having the line shipping cost in the “customer returns” tab like it is done for products. 
![screen shot on 2018-03-14 at 15_29_14](https://user-images.githubusercontent.com/824936/37409164-e9b8c886-279d-11e8-8759-18130d043377.png)

Do you think this way would be possible ? and/or a good thing to add to solidus ? I could probably work on a PR for that.
As noted by @gmacdougall, it doesn't really fit in to the customer return model, since they can't return the shipment...

I'd like to have opinions on that before starting anything.
Thanks.

**Comment by fkoessler** [comment]
Hello, I'd like to follow up on this issue and am available to work on a PR to allow shipping cost refund.

Since a Shipment is not something one can return, we could work with a Reimbursement instance that has no associated ReturnItems?
We should somehow link the Reimbursement to the Shipment though. This could be done by adding an optional relation between Reimbursements and Shipments. How does that sound?

From the UX point of view, should we add a new "Refund" tab that allows refunding all kind of non returnable items ? (I can only think of shipments for the moment) or should we add a line for shipping costs in the Customer return tab as shown in @loicginoux 's screenshot?
I think both options have their pros and cons.

I'm interested in your feedback.

**Comment by fkoessler** [comment]
I've been working on shipping costs refunds, here's where I stand so far:

### Models

I added a new `Settlement` model.
It is similar to the `ReturnItem` model.
It is used to store how much we want to refund for a shipment (as well as keeping track of status).
It could potentially be used later to store how much to refund for something else than a shipment, as long as it's not a return item.

Here's a class diagram that shows the associations:

![class-diagram](https://user-images.githubusercontent.com/1450340/40795008-0e99c614-6501-11e8-87b4-3417b1e95320.png)

### UX

It doesn't make sense to edit the settlements when picking items to return (shipments are not things you can return).
That's why I suggest to allow creating settlement when editing the reimbursement.
The UX could look like this:

![screenshot-localhost 3000-2018-05-31-18-01-16](https://user-images.githubusercontent.com/1450340/40795046-1d3864a0-6501-11e8-8f68-7728e69d3cc6.png)


Please share your feedbacks.
Would this new feature be considered for merging?

**Comment by dividedharmony** [comment]
If the shipping costs are taxed, is there a good way to display that information on this interface?

**Comment by fkoessler** [comment]
Similarly to the customer returns table, the shipments table could have a "Amount before tax" column as well as a "Total" column, instead of "Amount" and "Total":

![screenshot-localhost 3000-2018-06-04-16-19-51](https://user-images.githubusercontent.com/1450340/40922724-318b86de-6813-11e8-917a-2db41d3ff978.png)
